design-documents/web-api-controllers.md

```diff
@@ -31,6 +31,28 @@ doesn't make sense in context of a customer managing itself since a customer can
 All of these problems stem from having business layer contracts/entities exposed as presentation layer operations/data.
  
  
+###### Examples of classes that contain presentation layer logic
+While service contracts implementations usually can contain all the logic needed to execute an operation various
+presentation areas would require additional logic to convey results to users or to extract operations' arguments from
+user input. Since service contracts are not supposed to know about context they are also not feet to contain additional
+authorization related logic.
+ 
+Examples of cases when we need additional context aware/presentation logic:
+* When we need to validate entity ownership before performing an operation using authenticated user entity
+* Limit frequency of operation execution (for anti-bruteforce measures)
+* Prepare service layer DTOs for presentation, use additional service contracts for reading data to add to an
+  operation's result (most used in HTML area)
+* Plugin that would execute an additional operation based on session data/user context
+* Request throttling for a specific operation
```

Comment:
Why is this use case separate from anti brute force one?

Reply:
brute force involves captcha, the throttling involves managing response times